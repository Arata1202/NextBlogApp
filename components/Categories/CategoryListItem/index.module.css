.tag {
  border-radius: var(--border-radius);
  font-weight: bold;
  font-size: 15px !important;
  white-space: nowrap;
}
