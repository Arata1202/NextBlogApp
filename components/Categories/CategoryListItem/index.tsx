import React from 'react';
import { Category } from '@/libs/microcms';
import styles from './index.module.css';

type Props = {
  category: Category;
  hasLink?: boolean;
};

export default function CategoryListItem({ category, hasLink = true }: Props) {
  const handleClick = (event: React.MouseEvent) => {
    event.preventDefault();
    window.location.href = `/category/${category.id}`;
  };
  if (hasLink) {
    return (
      <a onClick={handleClick} href={`/category/${category.id}`} className={styles.tag}>
        <p className={styles.tag}>{category.name}</p>
      </a>
    );
  }
  return <span className={styles.tag}>{category.name}</span>;
}
