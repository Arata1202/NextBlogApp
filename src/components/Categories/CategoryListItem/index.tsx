import { Category } from '@/libs/microcms';
import styles from './index.module.css';

type Props = {
  category: Category;
  hasLink?: boolean;
};

export default function CategoryListItem({ category, hasLink = true }: Props) {
  if (hasLink) {
    return (
      <>
        <a href={`/category/${category.id}`}>
          <div className={styles.category}>{category.name}</div>
        </a>
      </>
    );
  }
  return (
    <>
      <div className={styles.category}>{category.name}</div>
    </>
  );
}
