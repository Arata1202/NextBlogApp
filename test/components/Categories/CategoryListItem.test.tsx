import { render } from '@testing-library/react';
import CategoryListItem from '@/components/Categories/CategoryListItem';
import type { Tag } from '@/libs/microcms';

describe('CategoryListItem', () => {
  test('CategoryListItem)', () => {
    const tag: Tag = { name: 'example', id: '1', createdAt: '', updatedAt: '' };
    const { asFragment } = render(<CategoryListItem tag={tag} hasLink={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
