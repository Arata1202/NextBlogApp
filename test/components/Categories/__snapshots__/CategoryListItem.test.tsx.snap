// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem スナップショット(CategoryListItem) 1`] = `
<DocumentFragment>
  <a
    class="tag"
    href="/category/1"
  >
    <p
      class="tag"
    >
      example
    </p>
  </a>
</DocumentFragment>
`;
